sdk/java/pulumi/src/test/java/com/pulumi/resources/StackTest.java

```diff
@@ -1,49 +1,51 @@
 package com.pulumi.resources;
 
+import com.pulumi.Context;
 import com.pulumi.core.Output;
 import com.pulumi.core.OutputTests;
 import com.pulumi.core.Tuples;
 import com.pulumi.core.Tuples.Tuple2;
-import com.pulumi.core.annotations.Export;
-import com.pulumi.core.internal.Internal;
 import com.pulumi.deployment.MocksTest;
+import com.pulumi.deployment.internal.DeploymentTests;
 import com.pulumi.deployment.internal.TestOptions;
-import com.pulumi.exceptions.RunException;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
 
 import java.util.Map;
-import java.util.function.Supplier;
+import java.util.function.Consumer;
+import java.util.function.Function;
 
 import static com.pulumi.deployment.internal.DeploymentTests.DeploymentMockBuilder;
 import static com.pulumi.deployment.internal.DeploymentTests.cleanupDeploymentMocks;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 
 class StackTest {
 
-    private static class ValidStack extends Stack {
+    private static class ValidStack {
+        public final Output<String> explicitName;
+        public final Output<String> implicitName;
 
-        @Export(name = "foo", type = String.class)
-        private final Output<String> explicitName;
+        public ValidStack(DeploymentTests.DeploymentMock.TestAsyncResult result) {
+            this.explicitName = result.getStackOutput("foo");
+            this.implicitName = result.getStackOutput("implicitName");
+       }
 
-        @Export(type = String.class)
-        private final Output<String> implicitName;
-
-        public ValidStack() {
-            this.explicitName = Output.of("bar");
-            this.implicitName = Output.of("buzz");
+        public static void init(Context ctx) {
+            ctx.export("foo", Output.of("bar"));
+            ctx.export("implicitName", Output.of("buzz"));
         }
     }
 
     @Test
     void testValidStackInstantiationSucceeds() {
-        var result = run(ValidStack::new);
-        assertThat(result.t2).hasSize(3);
+        var result = run(ValidStack::init, ValidStack::new);
+        assertThat(result.t2).hasSize(2);
```

Comment:
I was confused why hasSize(3) was there before. Sounds like only two outputs are there in the stack.